### DISCUSSION TITLE
Contract verification from remix zksync plugin isnt co-operating

### DISCUSSION THREAD

**Discussion by Tarreef** [Cyfrin/foundry-full-course-cu#2726] [Q&A]
when i try to verify and deploy to the zksync plugin wallet option,  get this error... 
>>> "Error in plugin @matterlabs/hardhat-zksync-verify: The address provided as argument contains a contract, but its bytecode doesn't match any of your local contracts."
Please, i'm stuck......

**Comment by EngrPips** [comment]
Hello @Tarreef, The below is the solution that worked someone who ran into similar issue 

```js
Hey Vishal! I know this is an old question, but I ran into the same issue and wanted to share my solution in case it helps anyone else.

After doing some research on Phind, I found that the issue often comes down to version compatibility between the ZKSync plugin and the current version of the Remix IDE or Solidity compiler.

What worked for me was changing the version of the Solidity compiler. In your SimpleStorage.sol file, you can use:

pragma solidity ^0.8.24;

The caret (^) allows any version greater than or equal to 0.8.24 but less than 0.9.0, which ensures you're staying within the same major version (0.8.x). This avoids breaking changes that might come with a major update like 0.9.x or beyond.

After that, you’ll need to update the version in the Solidity Compiler tab in Remix, recompile the contract, and recompile again in the ZKSync plugin. Once that’s done, you should be able to check the "Verify Contract" box and both deploy and verify the contract successfully.

Hope this helps you or anyone else in the future!

Cheers!
```

**Reply by EngrPips** [reply]
Also you should ensure that you are working in the `default-workspace` workspace and also ensure that your solidity files are inside the contracts folder.

**Comment by mitramansouri** [comment]
Hi there, I ran into the same error because many files were inside my remix default_workspace.
I deleted all the files ( except SimpleStorage.sol, of course ) and then created a folder called 'contracts' and put the SimpleStorage.sol file in there. I recompiled, hit deploy, and verified, and it worked.

**Reply by EngrPips** [reply]
Thanks for this heads-up
